### PR TITLE
issue: 1540484 Add UDP Rx Hardware timestamp support for upstream

### PR DESCRIPTION
**Pull request by liranoz12**
Only for mlx5 and above devices.

Signed-off-by: Liran Oz <lirano@mellanox.com>

### CODE REVIEW

**Comment by liranoz12** [comment]
bot:retest

**Comment by liranoz12** [comment]
bot:retest

File: config/m4/verbs.m4 [hunk 1]
```diff
@@ -27,7 +27,12 @@ AC_DEFUN([CHECK_VERBS_ATTRIBUTE], [
     ])
 ])
 
-
+# Check attributes
+# Usage: CHECK_VERBS_MEMBER([attribute], [header file], [definition])
+#
+AC_DEFUN([CHECK_VERBS_MEMBER], [
+    AC_CHECK_MEMBER( $1, [AC_DEFINE_UNQUOTED([DEFINED_$3], [1], [Define to 1 if attribute $1 is supported])], [], [#include <$2>])
+])
```
Comment (igor-ivanov) [inline config/m4/verbs.m4]: Probably you need to use square brackets to close header file (see: https://www.gnu.org/software/autoconf/manual/autoconf-2.64/html_node/Generic-Structures.html)
```
AC_DEFUN([CHECK_VERBS_MEMBER], [
    AC_CHECK_MEMBER( $1, [AC_DEFINE_UNQUOTED([DEFINED_$3], [1], [Define to 1 if attribute $1 is supported])], [], [[#include <$2>]])
])```

File: src/vma/ib/base/verbs_extra.h [hunk 2]
```diff
@@ -135,9 +135,19 @@ typedef struct ibv_qp_init_attr                     vma_ibv_qp_init_attr;
 #endif
 
 //ibv_query_device
-#define vma_ibv_query_device(context, attr)	ibv_query_device(context, attr)
-typedef struct ibv_device_attr			vma_ibv_device_attr;
-#define vma_ibv_device_attr_comp_mask(attr)	NOT_IN_USE(attr)
+#define vma_ibv_device_attr_comp_mask(attr)   NOT_IN_USE(attr)
+typedef struct ibv_device_attr                vma_ibv_device_attr;
```
Comment (igor-ivanov) [inline src/vma/ib/base/verbs_extra.h]: May be remove this because it is not used

**Comment by liranoz12** [comment]
@igor-ivanov updated.

**Comment by liranoz12** [comment]
bot:retest

**Pull request merged**